### PR TITLE
fix: avoid dbtRunner default callbacks being shared across instances 

### LINKED ISSUE #7278: [CT-2375] [Bug] dbtRunner instances share the callbacks across instances when no callbacks are passed
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

```py
from dbt.cli.main import dbtRunner, list
from dbt.events.base_types import EventMsg

def print_callback(event: EventMsg):
    print("Got an event:", event.info.name)

dbt = dbtRunner()
dbt.callbacks.append(print_callback)

dbt2 = dbtRunner()
dbt2.invoke(['list'])   # this prints the event names
```

example run:

```
❯ python runner.py
16:32:42  Running with dbt=1.5.0-b5
Got an event: MainReportVersion
Got an event: MainReportArgs
16:32:42  [WARNING]: Deprecated functionality
The `target-path` config in `dbt_project.yml` has been deprecated, and will no
longer be supported in a future version of dbt-core. If you wish to write dbt
artifacts to a custom directory, please use the --target-path CLI flag or
DBT_TARGET_PATH env var instead.
Got an event: ConfigTargetPathDeprecation
Got an event: SendingEvent
Got an event: SendingEvent
Got an event: SendingEvent
Got an event: StateCheckVarsHash
Got an event: PartialParsingEnabled
Got an event: PartialParsingSkipParsing
Got an event: SendingEvent
Got an event: SendingEvent
16:32:42  Found 4 models, 0 tests, 0 snapshots, 0 analyses, 309 macros, 0 operations, 0 seed files, 0 sources, 0 exposures, 0 metrics, 0 groups
Got an event: FoundStats
Got an event: SendingEvent
simple_test.base.model_a
simple_test.base.model_b
simple_test.python.model_c
simple_test.model_d
Got an event: CommandCompleted
Got an event: SendingEvent
Got an event: FlushEvents
```

### Expected Behavior

dbtRunner instances should be isolated from one another.

### Steps To Reproduce

See current behavior

### Relevant log output

_No response_

### Environment

```markdown
- OS: macOS Ventura
- Python: Python 3.8.13
- dbt: 1.5.0-b5
```

### Which database adapter are you using with dbt?

postgres

### Additional Context

https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m

### PR DESCRIPTION
**Pull request by chamini2**
resolves #7278

### Description

default the callbacks argument of dbtRunner to None and create a new list as the default every time it runs.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/cli/main.py [hunk 1]
```diff
@@ -40,11 +40,14 @@ def __init__(
         project: Project = None,
         profile: Profile = None,
         manifest: Manifest = None,
-        callbacks: List[Callable[[EventMsg], None]] = [],
+        callbacks: List[Callable[[EventMsg], None]] = None,
```
Comment (jtcohen6) [inline core/dbt/cli/main.py:43]: ah - reminds me of an all-time classic: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Comment (chamini2) [reply core/dbt/cli/main.py:43]: We have all been caught by it at some point!

**Review by QMalcolm — APPROVED** [review]
Nice! Thanks for catching this and also going about fixing it 🙂

**Review by McKnight-42 — APPROVED** [review]

**Comment by jtcohen6** [comment]
Thanks @chamini2! I'm going to merge this, since it already has approval

**Pull request merged**